libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -901,7 +904,7 @@ void AC_PosControl::update_pos_offset_z(float pos_offset_z)
 // is_active_z - returns true if the z position controller has been run in the previous 5 loop times
 bool AC_PosControl::is_active_z() const
 {
-    return ((AP_HAL::micros64() - _last_update_z_us) <= _dt * 5000000.0);
+    return ((AP::ins().get_last_update_usec() - _last_update_z_us) <= _dt* 2000000.0);
```

Comment:
replacing the AP_HAL::micros with the last INS update time is a nice enhancement but is actually unrelated to the main purpose of this PR.   We have millions of places where we make unnecessary calls to AP_HAL to ask the time.

Reply:
Not really. AP::ins().get_last_update_usec() is the equivlent of AP_HAL::micros64() given we are referencing the time we run this controller to the last gyro read. In any normal cycle we expect:
(AP::ins().get_last_update_usec() - _last_update_z_us) = _dt

So any increase over this means we have missed running this in a cycle. I have increased this back up to 5 times but I think it should always be ==